### ISSUE TITLE
some issue about training the network

### ISSUE THREAD

**Issue by qxiaomay** [rabi3elbeji/SteganalysisCNN#1]
dear author:
I have a question when training this network:
About your database, How to create it? How to preprocess it?

best Regards

**Comment by rabi3elbeji** [comment]
Hello,
I used the [Bossbase dataset](http://agents.fel.cvut.cz/stegodata/).
First, I resized all the images at 128x128 with cubic interpolation then I applied the different embedding to the resized images.
A Preprocessing was included on the first layer of the network using non-trainable [HPF filters](https://github.com/rabi3elbeji/SteganalysisCNN/blob/master/train.py#L105).
[Paper](https://drive.google.com/file/d/15B7Nc6XhrIKhtNuOt4LlPEauW_gtekwr/view?usp=sharing).

**Comment by qxiaomay** [comment]
Thanks for your reply~
I wonder what the data format of the images you used.
Have you converted the image format?
The Bossbase dataset images are in '.pgm' format, and I found that Keras ImageDatagenerator reads others file-formats but does not read '.pgm'.

best Regards

**Comment by rabi3elbeji** [comment]
Yes, for the PGM format, I have the same problem.
I was able to solve this problem by adding the .pgm format to the whitelist in ImageDataGenerator class.

**Comment by rabi3elbeji** [comment]
Sorry, not in the **ImageDataGenerator** class but in the class `image.py` (package `keras\preprocessing`) 
Add `pgm` to this white list (line 974)
```python
white_list_formats = {'png', 'jpg', 'jpeg', 'bmp', 'ppm'}
```

**Comment by qxiaomay** [comment]
Thanks you very much! I've solved the data format problem.
I use the BOSSbase_1.01 dataset, resize the images at 128*128, embedding with WOW 1.0bpp
I trained the network, and I got train acc 0.78, loss 0.38.
I am very confused, may I ask what went wrong?

**Comment by rabi3elbeji** [comment]
Are you changed [the number of epochs](https://github.com/rabi3elbeji/SteganalysisCNN/blob/master/train.py#L63) for 1.0bpp to 100! 

![wow_1bpp](https://user-images.githubusercontent.com/12187724/59548709-91a3cf00-8f53-11e9-9e44-4f792fc963a6.PNG)

Also, I use only the first **2400 images**.

![dataset](https://user-images.githubusercontent.com/12187724/59548724-b6984200-8f53-11e9-832f-d812e2497091.PNG)

Could you please send me a screenshot of training epochs?

**Comment by qxiaomay** [comment]
Yes, I have changed the number of epochs, I use 2500 images random selected from BOSSbase_1.01 dataset.
![1560735315(1)](https://user-images.githubusercontent.com/24953690/59572987-3dc7f000-90e3-11e9-9cd8-56e67f8a9ad5.jpg)

Here is the screenshot of training epochs.
![screenshot](https://user-images.githubusercontent.com/24953690/59572947-022d2600-90e3-11e9-8229-1f7a9102379a.jpg)

There must be something wrong〒▽〒.

Looking forward to your reply!

**Comment by sararaza534** [comment]
Is it possible to run the code on windows?

**Comment by sararaza534** [comment]
can you please guide what files to execute?

**Comment by qxiaomay** [comment]
I run the code on linux. just excute the 'train.py' and edit the parameter.

**Comment by HgKhan47** [comment]
>
(https://user-images.githubusercontent.com/59117065/72331088-72495d80-366c-11ea-90c4-7a92da005580.png)
Respected Author , I have one last issue .png file is attached below . I already make folders for training and validation set but i don`t know why this problem still there.

**Comment by silent-code** [comment]
@HgKhan47: see the comment above on modifying the whitelist in Keras:

"in the class image.py (package keras\preprocessing)
Add pgm to this white list (line 974)

white_list_formats = {'png', 'jpg', 'jpeg', 'bmp', 'ppm'}"

NOTE: This doesn't appear to work in later versions of Keras, e.g., 2.3 because the implementation has changed.

Also see this github issue: https://github.com/keras-team/keras/issues/7904

**Comment by HgKhan47** [comment]
@silent-code: i have already done that, even then the images in two classes are not found. i still didn't figure out what is the reason behind it while setting up the maximum paths for it too!!

**Comment by silent-code** [comment]
Hi @rabi3elbeji, I am achieving similar training accuracy as your paper reports: 

Epoch 100/100
 loss: 0.1259 - acc: 0.9560 - val_loss: 2.0798 - val_acc: 0.4925

However my validation accuracy remains around 0.5.

I noticed you do not report a validation or test accuracy in your paper. Care to comment?

Thanks, Tim
